src/backend/aspen/api/utils/sample.py

```diff
@@ -1,15 +1,46 @@
 from collections import Counter
-from typing import Any, Dict, List, Mapping, Optional, Tuple, TYPE_CHECKING
+from typing import Any, Dict, List, Mapping, Optional, Set, Tuple, TYPE_CHECKING
 
 import sqlalchemy as sa
 from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.orm.query import Query
+from sqlalchemy.sql.expression import or_
 
+from aspen.api.authz import AuthZSession
 from aspen.database.models import Accession, AccessionType, Sample
 
 if TYPE_CHECKING:
     from aspen.api.schemas.samples import CreateSampleRequest
 
 
+async def samples_by_identifiers(
+    az: AuthZSession, sample_ids: Optional[Set[str]]
+) -> Query:
```

Comment:
Where we moved the "get samples" code from the authz module

Reply:
this looks much cleaner!